Parsers/ASimAuthentication/Parsers/ASimAuthenticationCiscoMeraki.yaml

```diff
@@ -0,0 +1,103 @@
+Parser:
+  Title: ASIM Authentication parser for Cisco Meraki
+  Version: '0.1'
+  LastUpdated: Jun 12 2023
+Product:
+  Name: Cisco Meraki
+Normalization:
+  Schema: Authentication
+  Version: '0.1.3'
+References:
+- Title: ASIM Authentication Schema
+  Link: https://aka.ms/ASimAuthenticationDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+Description: |
+  This ASIM parser supports normalizing the Cisco Meraki logs to the ASIM authentication normalized schema.
+ParserName: ASimAuthenticationCiscoMeraki
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+
+ParserQuery: |
+  let EventResultDetailsLookup = datatable (reason:string, EventResultDetails:string)
+  [
+      "0","Other",
+      "1","Other",
+      "2","Password expired",
+      "3","Other",
+      "4","Session expired",
+      "5","Other",
+      "6","Other",
+      "7","Other",
+      "8","Other",
+      "9","Other",
+      "10","Logon violates policy",
+      "11","Logon violates policy",
+      "12","Other",
+      "13","Logon violates policy",
+      "14","Other",
+      "15","Other",
+      "16","Other",
+      "17","Other",
+      "18","Incorrect key",
+      "19","Incorrect key",
+      "20","Incorrect key",
+      "21","Other",
+      "22","Other",
+      "23","Other",
+      "24","Logon violates policy",
+  ];
+  let parser = (disabled:bool=false){
+    union isfuzzy=true
+          (
+        meraki_CL
+        | project-rename LogMessage =  Message
+          ),
+          (
+        Syslog
+        | project-rename LogMessage =  SyslogMessage
+          )
+      | where not(disabled) and LogMessage has "events" and (LogMessage has_any ("8021x_auth", "wpa_auth", "splash_auth", "8021x_deauth", "8021x_client_deauth", "wpa_deauth", "8021x_eap_failure", "8021x_eap_success") or (LogMessage has "disassociation" and LogMessage has "auth_neg_failed"))
+      | extend Parser = extract_all(@"(\d+.\d+)\s([\w\-\_]+)\s([\w\-\_]+)\s([\S\s]+)$", dynamic([1, 2, 3, 4]), LogMessage)[0]
```

Comment:
Since all the literals are space separated, parse can be used which would eliminate the need for regex.

Reply:
The event format are not of the same type so we are not able to implement it.

**Event format1:**
2019-07-16 04:00:30 [Local0.Info](http://local0.info/) 172.40.20.177 1 1561036264.565291108 australia_sydney security_event security_filtering_file_scanned url=http://tlu123.dl123.delivery.mp.microsoft.com/filestreamingservice/files/36ec4eb7-46dd-4aeb-990e-b6d32f7ed567?P1=1561036860&amp; P2=402&amp;P3=2&amp;P4=Zj3qRDR5CbzfWlP8BuYg%2bUlTon0XE774ExEEquiawstLAJ2%2bQm3OoWLcwz3HBt8qp3r3buVRVoT5BQcUCcNlXw%3d%3d src=[172.16.10.98:64160](http://172.16.10.98:64160/) dst=[200.188.210.42:180](http://200.188.210.42:180/) mac=20:1C:BC:B2:0F:20 name='' sha256093e4fc218b27e58e2fede7b8cb044d48d66995ae785bbc186a9df5ae08ca4f7 disposition=malicious action=block

**Event format2:**
<134>1 1685956046.281934865 CDSAPF3_11 events type=association radio='1' vap='3' client_mac='F2:ED:56:6A:98:6A' last_known_client_ip='172.20.12.194' band='5' channel='104' rssi='38' identity='zuber.chataiwala' aid='1305860724'

Reply:
| parse LogMessage with * TimeEpoch "." * " " Hostname " " LogType " " Substring

After this, you can divide the code blocks and perform parsing. Combine them back with union.

Reply:
We double checked and we have the different event formats as above so we are not able to do the parsing as above to get the required fields.

In authentication all events are of same type so no need for the code blocks.

Reply:
Please check the WebSession schema. I replied on it. Thanks